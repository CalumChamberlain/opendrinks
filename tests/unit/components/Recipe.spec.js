import { createLocalVue, mount } from '@vue/test-utils';
import BootstrapVue from 'bootstrap-vue';

import router from '@/router';
import Recipe from '@/components/Recipe.vue';

const localVue = createLocalVue();

localVue.use(BootstrapVue);

describe('Recipe', () => {
  const wrapper = mount(Recipe, {
    propsData: { name: 'mango-juice.json' },
    localVue,
    router,
  });

  test('is a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  test('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
