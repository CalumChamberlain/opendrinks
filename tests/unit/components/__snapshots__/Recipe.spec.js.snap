// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipe renders correctly 1`] = `
<div
  id="recipe"
>
  <div
    class="print-button"
  >
    <a
      class="btn float-right btn-outline-primary"
      href="/recipe/mango-juice.json/print"
      rel="noopener"
      target="_blank"
    >
      
      Print
    
    </a>
  </div>
   
  <h1>
    Mango Juice
  </h1>
   
  <div
    class="mb-2"
  >
    <a
      class="badge badge-secondary"
      href="/keyword/non-alcoholic"
      style="margin-right: 0.2vw;"
      target="_self"
    >
      non-alcoholic
    </a>
    <a
      class="badge badge-secondary"
      href="/keyword/vegan"
      style="margin-right: 0.2vw;"
      target="_self"
    >
      vegan
    </a>
    <a
      class="badge badge-secondary"
      href="/keyword/smoothie"
      style="margin-right: 0.2vw;"
      target="_self"
    >
      smoothie
    </a>
  </div>
   
  <p>
    Mango is a pulpy fruit which requires more liquids when blended to juice. Mango juice can do wonders to your body especially for the digestion.
  </p>
   
  <svg
    class="float-right"
    enable-background="new 0 0 24 24"
    height="28px"
    id="star"
    version="1.1"
    viewBox="0 0 24 24"
    width="28px"
    x="0px"
    xml:space="preserve"
    xmlns="http://www.w3.org/2000/svg"
    xmlns:xlink="http://www.w3.org/1999/xlink"
    y="0px"
  >
    <g
      id="Bounding_Boxes"
    >
      <path
        d="M0,0h24v24H0V0z"
        fill="none"
      />
       
      <path
        d="M0,0h24v24H0V0z"
        fill="none"
      />
    </g>
     
    <g
      fill="white"
      id="Outline_1_"
      stroke="black"
    >
      <path
        class=""
        d="M12,17.27L18.18,21l-1.64-7.03L22,9.24l-7.19-0.61L12,
      2L9.19,8.63L2,9.24l5.46,4.73L5.82,21L12,17.27z"
      />
    </g>
  </svg>
   
  <p
    class="text-muted"
  >
    
    Contributed by:
    
    <a
      href="https://github.com/Prem1835"
    >
      Prem1835
    </a>
  </p>
   
  <img
    alt="Mango Juice"
    class="mb-4 img-fluid w-100 rounded"
    src=""
  />
   
  <h4>
    Ingredients
  </h4>
   
  <ul>
    <li>
      
      1  large mango
    
    </li>
    <li>
      
      2 cups water or coconut water
    
    </li>
    <li>
      
      5 to 6  mint leaves
    
    </li>
    <li>
      
      1 cup of water
    
    </li>
  </ul>
   
  <h4>
    Directions
  </h4>
   
  <ol
    class="mb-4"
  >
    <li>
      Wash mango thoroughly and chill. Peel the skin.
    </li>
    <li>
      Cut the mango into cubes and add them to a blender and pour the coconut water or orange juice
    </li>
    <li>
      Put in the other ingredients if using like ginger and mint
    </li>
    <li>
      Blend until smooth. Serve immediately.
    </li>
  </ol>
   
  <!---->
   
  <div
    class="similarDrinks mt-4 mb-4"
  >
    <h4>
      Similar drinks
    </h4>
     
    <div
      class="card-deck"
    >
      <div
        class="card mb-4"
        href="/recipe/orange-glorius"
        style="min-width: calc( 50% - 30px );"
      >
        <!---->
        <!---->
        <a
          class=""
          href="/recipe/orange-glorius"
          target="_self"
        >
          <img
            alt="Orange Glorious"
            class="card-img-top"
            src=""
          />
        </a>
         
        <div
          class="card-body"
        >
          <!---->
          <!---->
          <h4
            class="card-title"
          >
            <a
              class=""
              href="/recipe/orange-glorius"
              target="_self"
            >
              <h5>
                Orange Glorious
              </h5>
            </a>
          </h4>
           
          <p
            class="card-text"
          >
            
      Special Orange Juice Mixed latte this beverage tas ...
    
          </p>
           
          <p
            class="card-text"
          >
            <div>
              <a
                class="badge mr-1 badge-primary"
                href="/keyword/smoothie"
                target="_self"
              >
                
          smoothie
        
              </a>
              <a
                class="badge mr-1 badge-primary"
                href="/keyword/orange"
                target="_self"
              >
                
          orange
        
              </a>
              <a
                class="badge mr-1 badge-primary"
                href="/keyword/milk"
                target="_self"
              >
                
          milk
        
              </a>
              <a
                class="badge mr-1 badge-primary"
                href="/keyword/vegan"
                target="_self"
              >
                
          vegan
        
              </a>
              <a
                class="badge mr-1 badge-primary"
                href="/keyword/non-alcoholic"
                target="_self"
              >
                
          non-alcoholic
        
              </a>
            </div>
          </p>
        </div>
        <!---->
        <!---->
      </div>
      <div
        class="card mb-4"
        href="/recipe/mix-berry"
        style="min-width: calc( 50% - 30px );"
      >
        <!---->
        <!---->
        <a
          class=""
          href="/recipe/mix-berry"
          target="_self"
        >
          <img
            alt="Mix Berry Juice"
            class="card-img-top"
            src=""
          />
        </a>
         
        <div
          class="card-body"
        >
          <!---->
          <!---->
          <h4
            class="card-title"
          >
            <a
              class=""
              href="/recipe/mix-berry"
              target="_self"
            >
              <h5>
                Mix Berry Juice
              </h5>
            </a>
          </h4>
           
          <p
            class="card-text"
          >
            
      Berries are fresh fruit and can be processed into  ...
    
          </p>
           
          <p
            class="card-text"
          >
            <div>
              <a
                class="badge mr-1 badge-primary"
                href="/keyword/non-alcoholic"
                target="_self"
              >
                
          non-alcoholic
        
              </a>
              <a
                class="badge mr-1 badge-primary"
                href="/keyword/vegan"
                target="_self"
              >
                
          vegan
        
              </a>
              <a
                class="badge mr-1 badge-primary"
                href="/keyword/smoothie"
                target="_self"
              >
                
          smoothie
        
              </a>
              <a
                class="badge mr-1 badge-primary"
                href="/keyword/milk"
                target="_self"
              >
                
          milk
        
              </a>
              <a
                class="badge mr-1 badge-primary"
                href="/keyword/blueberries"
                target="_self"
              >
                
          blueberries
        
              </a>
              <a
                class="badge mr-1 badge-primary"
                href="/keyword/strawberries"
                target="_self"
              >
                
          strawberries
        
              </a>
            </div>
          </p>
        </div>
        <!---->
        <!---->
      </div>
      <div
        class="card mb-4"
        href="/recipe/kiba-smoothie"
        style="min-width: calc( 50% - 30px );"
      >
        <!---->
        <!---->
        <a
          class=""
          href="/recipe/kiba-smoothie"
          target="_self"
        >
          <img
            alt="Kiba - A german smoothie"
            class="card-img-top"
            src=""
          />
        </a>
         
        <div
          class="card-body"
        >
          <!---->
          <!---->
          <h4
            class="card-title"
          >
            <a
              class=""
              href="/recipe/kiba-smoothie"
              target="_self"
            >
              <h5>
                Kiba - A german smoothie
              </h5>
            </a>
          </h4>
           
          <p
            class="card-text"
          >
            
      Kiba is a smoothie made with bananas, cherries, an ...
    
          </p>
           
          <p
            class="card-text"
          >
            <div>
              <a
                class="badge mr-1 badge-primary"
                href="/keyword/smoothie"
                target="_self"
              >
                
          smoothie
        
              </a>
              <a
                class="badge mr-1 badge-primary"
                href="/keyword/fruits"
                target="_self"
              >
                
          fruits
        
              </a>
              <a
                class="badge mr-1 badge-primary"
                href="/keyword/vegan"
                target="_self"
              >
                
          vegan
        
              </a>
              <a
                class="badge mr-1 badge-primary"
                href="/keyword/non-alcoholic"
                target="_self"
              >
                
          non-alcoholic
        
              </a>
            </div>
          </p>
        </div>
        <!---->
        <!---->
      </div>
      <div
        class="card mb-4"
        href="/recipe/aam-panna"
        style="min-width: calc( 50% - 30px );"
      >
        <!---->
        <!---->
        <a
          class=""
          href="/recipe/aam-panna"
          target="_self"
        >
          <img
            alt="Aam Panna"
            class="card-img-top"
            src=""
          />
        </a>
         
        <div
          class="card-body"
        >
          <!---->
          <!---->
          <h4
            class="card-title"
          >
            <a
              class=""
              href="/recipe/aam-panna"
              target="_self"
            >
              <h5>
                Aam Panna
              </h5>
            </a>
          </h4>
           
          <p
            class="card-text"
          >
            
      Aam panna is an Indian drink renowned for its heat ...
    
          </p>
           
          <p
            class="card-text"
          >
            <div>
              <a
                class="badge mr-1 badge-primary"
                href="/keyword/fruit"
                target="_self"
              >
                
          fruit
        
              </a>
              <a
                class="badge mr-1 badge-primary"
                href="/keyword/juice"
                target="_self"
              >
                
          juice
        
              </a>
              <a
                class="badge mr-1 badge-primary"
                href="/keyword/mocktail"
                target="_self"
              >
                
          mocktail
        
              </a>
              <a
                class="badge mr-1 badge-primary"
                href="/keyword/non-alcoholic"
                target="_self"
              >
                
          non-alcoholic
        
              </a>
              <a
                class="badge mr-1 badge-primary"
                href="/keyword/vegan"
                target="_self"
              >
                
          vegan
        
              </a>
              <a
                class="badge mr-1 badge-primary"
                href="/keyword/mango"
                target="_self"
              >
                
          mango
        
              </a>
            </div>
          </p>
        </div>
        <!---->
        <!---->
      </div>
    </div>
  </div>
</div>
`;
